src/protocol/malicious.rs

```diff
@@ -245,52 +285,30 @@ pub mod tests {
 
         let futures =
             zip(context, zip(a_shares, b_shares)).map(|(ctx, (a_share, b_share))| async move {
-                let v = SecurityValidator::new(ctx.narrow("SecurityValidatorInit"));
-                let acc = v.accumulator();
-                let r_share = v.r_share();
-
-                let a_ctx = ctx.narrow("1");
-                let b_ctx = ctx.narrow("2");
-
-                let (ra, rb) = try_join(
-                    a_ctx
-                        .narrow("input")
-                        .multiply(RecordId::from(0), &a_share, r_share),
-                    b_ctx
-                        .narrow("input")
-                        .multiply(RecordId::from(0), &b_share, r_share),
-                )
-                .await?;
-
-                let a_ctx = a_ctx.upgrade_to_malicious(acc.clone(), r_share.clone());
-                let b_ctx = b_ctx.upgrade_to_malicious(acc.clone(), r_share.clone());
+                let v = MaliciousValidator::new(ctx);
+                let m_ctx = v.context();
 
-                let a_malicious = MaliciousReplicated::new(a_share, ra);
-                let b_malicious = MaliciousReplicated::new(b_share, rb);
-
-                acc.accumulate_macs(
-                    &a_ctx.narrow(&Step::ValidateInput).prss(),
-                    RecordId::from(0),
-                    &a_malicious,
-                );
-                acc.accumulate_macs(
-                    &b_ctx.narrow(&Step::ValidateInput).prss(),
-                    RecordId::from(0),
-                    &b_malicious,
-                );
+                let a_malicious = m_ctx.upgrade(RecordId::from(0), a_share).await?;
+                let b_malicious = m_ctx.upgrade(RecordId::from(1), b_share).await?;
 
-                let mult_result = a_ctx
+                let m_result = m_ctx
                     .multiply(RecordId::from(0), &a_malicious, &b_malicious)
                     .await?;
 
+                // Save some cloned values so that we can check them.
                 let r_share = v.r_share().clone();
-                v.validate(ctx.narrow("SecurityValidatorValidate")).await?;
-                Ok::<_, Error>((mult_result, r_share))
+                let result = v.validate(m_result.clone()).await?;
```

Comment:
I don't understand why we need to pass in the multiplication result to the validate function. Validation is over the entire protocol thus far, and we do not necessarily want to reveal anything when we validate a protocol.

Reply:
This is something that Alex and I discussed as a code hygiene thing.  This forces people who have results in `MaliciousReplicated` form to call `validate()` before they can access the results.  Otherwise, there is nothing stopping them from releasing the value of `m_share.x()` without waiting for the validation to complete.